### PR TITLE
fix(service-worker): handle error with ErrorHandler

### PR DESCRIPTION
**Pull request by chrisguttandin**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #39913

## What is the new behavior?

Errors thrown when trying to register a Service Worker are now passed to the global ErrorHandler.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/service-worker/src/module.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@
  */
 
 import {isPlatformBrowser} from '@angular/common';
-import {APP_INITIALIZER, ApplicationRef, InjectionToken, Injector, ModuleWithProviders, NgModule, NgZone, PLATFORM_ID} from '@angular/core';
+import {APP_INITIALIZER, ApplicationRef, InjectionToken, Injector, ErrorHandler, ModuleWithProviders, NgModule, NgZone, PLATFORM_ID} from '@angular/core';
```
Comment (gkalpak) [inline packages/service-worker/src/module.ts:10]:
```suggestion
import {APP_INITIALIZER, ApplicationRef, ErrorHandler, InjectionToken, Injector, ModuleWithProviders, NgModule, NgZone, PLATFORM_ID} from '@angular/core';
```

File: packages/service-worker/src/module.ts [hunk 2]
```diff
@@ -121,6 +121,7 @@ export function ngswAppInitializer(
       }
     }
 
+    const errorHandler = injector.get(ErrorHandler);
```
Comment (gkalpak) [inline packages/service-worker/src/module.ts:124]: I suggest we move this line into the `catch()` callback, so we avoid retrieving the `ErrorHandler` if we don't need it.

**Review by gkalpak — CHANGES_REQUESTED** [review]
Thx for working on this, @chrisguttandin :+1:

I've left a couple of minor comments. Could you also add a test for the new behavior in [service-worker/test/module_spec.ts][1]?

Also, let's add a short description of the motivation for the change in the commit message body and also add `Fixes #39913` at the bottom (per our [commit message guidelines][2]).

[1]: https://github.com/angular/angular/blob/63213a78cdc1f31cd13dd2e43c3a564168fa424d/packages/service-worker/test/module_spec.ts
[2]: https://github.com/angular/angular/blob/1d9e56eba06958e37ef8062ae7b65afa85f90632/CONTRIBUTING.md#commit

File: packages/service-worker/src/module.ts (hunk 1 above)
Comment (chrisguttandin) [reply packages/service-worker/src/module.ts]: Thanks for catching this. I meant to order the imports alphabetically but got it wrong.

File: packages/service-worker/src/module.ts (hunk 2 above)
Comment (chrisguttandin) [reply packages/service-worker/src/module.ts]: Yes, good idea. I applied it.

**Comment by chrisguttandin** [comment]
Hi @gkalpak, thanks for your feedback. I made the changes and updated the test. Please let me know if there is anything else I should change.

File: packages/service-worker/test/module_spec.ts [hunk 3]
```diff
@@ -34,9 +35,14 @@ describe('ServiceWorkerModule', () => {
...
   describe('register()', () => {
     const configTestBed = async (opts: SwRegistrationOptions) => {
+      const errorHandler = {handleError: () => {}};
+      errorHandlerSpy = spyOn(errorHandler, 'handleError');
       TestBed.configureTestingModule({
         imports: [ServiceWorkerModule.register('sw.js', opts)],
-        providers: [{provide: PLATFORM_ID, useValue: 'browser'}],
+        providers: [
+          {provide: ErrorHandler, useValue: errorHandler},
+          {provide: PLATFORM_ID, useValue: 'browser'}
```
Comment (gkalpak) [inline packages/service-worker/test/module_spec.ts:44]: Nit:

```suggestion
          {provide: PLATFORM_ID, useValue: 'browser'},
```

**Review by gkalpak — APPROVED** [review]
One super-minor nit. Otherwise lgtm (as long as CI is happy :smiley:)
Thx again, @chrisguttandin ✨

File: packages/service-worker/test/module_spec.ts (hunk 3 above)
Comment (chrisguttandin) [reply packages/service-worker/test/module_spec.ts]: No problem, I added the comma.

**Pull request closed without merging**